sklearn/multiclass.py

```diff
@@ -203,6 +213,49 @@ def fit(self, X, y):
 
         return self
 
+    def partial_fit(self, X, y, classes=None):
+        """Partially fit underlying estimators
+        
+        Should be used when memory is inefficient to train all data.
+        Chunks of data can be passed in several iteration, where the
+        first call should have an array of all target variables.
+        
+        Parameters
+        ----------
+        X : (sparse) array-like, shape = [n_samples, n_features]
+            Data.
+        y : (sparse) array-like, shape = [n_samples] or [n_samples, n_classes]
+            Multi-class targets. An indicator matrix turns on multilabel
+            classification.
+        Returns
+        -------
+        self
+        """
+        if self._first_partial_run:
+            if classes is None:
+                raise ValueError("array of all classes of target varialble"
+                                  "must be provided")
+            self.classes_ = np.unique(classes)
+            self._first_partial_run = False
+            self.estimators_ = [clone(self.estimator) for _ in range
+                                (self.n_classes_)]
+
+        self.label_binarizer_ = LabelBinarizer(sparse_output=True)
+        Y = self.label_binarizer_.fit_transform(y)
+        Y = Y.tocsc()
+        columns = (col.toarray().ravel() for col in Y.T)
```

Comment:
I don't really understand: why are we using a sparse_output for the LabelBinarizer and converting to arrays right after. I don't see the benefit of using a sparse output and densifying. Would it be simpler and more efficient to not use sparse_output on the binarizer and to do simple slicing to retrieve the column.